modis_tools/granule_handler.py

```diff
@@ -95,7 +93,9 @@ def _coerce_to_list(
         return possible_list
 
     @staticmethod
-    def get_url_from_granule(granule: Granule, ext: ParamType = "hdf") -> HttpUrl:
+    def get_url_from_granule(
+        granule: Granule, ext: Tuple = ("hdf", "h5", "nc", "xml")
```

Comment:
Personally, I would add ext to the `download_from_granules` method as well to make this more flexible for users.